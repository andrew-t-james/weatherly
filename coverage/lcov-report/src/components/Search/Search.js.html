<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Search/Search.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Search</a> Search.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.3% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>21/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.43% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.3% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>21/23</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">20x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">20000x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Trie from 'location-autocomplete-trie';
import { cities } from '../../data-helpers/cities.js';
import './Search.css';
&nbsp;
class Search extends Component {
  constructor() {
    super();
    this.state = {
      location: '',
      autoComplete: new Trie()
    };
&nbsp;
    this.state.autoComplete.populate(cities);
    this.state.autoComplete.suggest(this.state.location);
    this.updateInput = this.updateInput.bind(this);
    this.submitLocation = this.submitLocation.bind(this);
  }
&nbsp;
  updateInput(event) {
    const { name, value } = event.target;
&nbsp;
    this.setState({
      [name]: value
    });
  }
&nbsp;
  submitLocation(event) {
    const { location } = this.state;
    const sanitizedUserInput = location.split(', ');
    const city = sanitizedUserInput[0];
    const state = sanitizedUserInput[1];
&nbsp;
    event.preventDefault();
    this.props.updateLocation(city, state);
&nbsp;
    this.setState({
      location: ''
    });
&nbsp;
    event.target.reset();
  }
&nbsp;
  render() {
    const  { location } = this.state;
    const { hasError } = this.props;
&nbsp;
    return (
      &lt;header className="header"&gt;
        &lt;h1
          tabIndex="0"
          className="header__title"
          aria-label="Weathrly A weather App"&gt;
        Weathrly
        &lt;/h1&gt;
        &lt;form
          className="search-form"
          onSubmit={<span class="fstat-no" title="function not covered" >ev</span>ent =&gt; <span class="cstat-no" title="statement not covered" >this.submitLocation(event)}</span>&gt;
          &lt;input
            className={`search-form__input ${hasError ? 'error' : ''}`}
            placeholder="Search by City and State or Zip"
            onChange={<span class="fstat-no" title="function not covered" >ev</span>ent =&gt; <span class="cstat-no" title="statement not covered" >this.updateInput(event)}</span>
            name="location"
            type="text"
            list='cities'
            aria-label="Input a City and State, or Zip code to search for weather by location"
            value={location} /&gt;
          &lt;datalist id="cities"&gt;
            {
              cities.map((city, index) =&gt;
                &lt;option
                  key={index}
                  value={city} /&gt;
              )
            }
          &lt;/datalist&gt;
          &lt;button
            tabIndex="0"
            aria-label="Search for weather"
            className="search-form__button"
            type="submit"
            disabled={!location.trim(' ')}&gt;
            Search
          &lt;/button&gt;
        &lt;/form&gt;
      &lt;/header&gt;
    );
  }
}
&nbsp;
Search.propTypes = {
  updateLocation: PropTypes.func,
  hasError: PropTypes.bool,
  autoComplete: PropTypes.func
};
&nbsp;
export default Search;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 31 2018 09:58:02 GMT-0600 (MDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
