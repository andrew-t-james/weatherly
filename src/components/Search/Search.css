.header {
  align-items: center;
  display: flex;
  justify-content: space-around;
}

/* .search-form {
  align-items: center;
  display: flex;
  flex: 0 0 40%;
  justify-content: center;
}

.search-form__input {
  background-color: var(--color-grey-light-2);
  border: none;
  border-radius: 100px;
  color: inherit;
  font-family: inherit;
  font-size: inherit;
  margin-right: -5.25rem;
  padding: 0.7rem 2rem;
  transition: all 0.2s;
  width: 90%;
}

.search-form__input:focus {
  background-color: var(--color-grey-light-3);
  outline: none;
  width: 100%;
}

.search-form__input::-webkit-input-placeholder {
  color: var(--color-grey-light-4);
  font-weight: 100;
}

.search-form__input:focus + .search-form__button {
  background-color: var(--color-grey-light-3);
}

.search-form__button {
  background-color: var(--color-grey-light-2);
  border: none;
  text-transform: uppercase;
}

.search-form__button:focus {
  outline: none;
}

.search-form__button:hover {
  cursor: pointer;
}

.search-form__button:active {
  transform: translateY(2px);
}

input:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px white inset;
} */