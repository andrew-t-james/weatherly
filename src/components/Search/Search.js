import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Trie from '../../data-helpers/Trie.js';
import { cities } from '../../data-helpers/cities.js';

import './Search.css';

// TODO: Add Prefix trie to search component

class Search extends Component {
  constructor() {
    super();
    this.state = {
      location: '',
      autoComplete: new Trie()
    };

    this.state.autoComplete.populate(cities);
    this.state.autoComplete.suggest(this.state.location);
    this.updateInput = this.updateInput.bind(this);
    this.submitLocation = this.submitLocation.bind(this);
  }

  updateInput(event) {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  }

  submitLocation(event) {
    const { location } = this.state;
    const splitString = location.trim(' ').split(', ');
    const city = splitString[0];
    const state = splitString[1];

    event.preventDefault();
    this.props.updateLocation(city, state);

    this.setState({
      location: ''
    });

    event.target.reset();
  }

  render() {
    const  { location } = this.state;

    return (
      <header className="header">
        <h1 className="header__title">Weathrly</h1>
        <form
          className="search-form"
          onSubmit={event => this.submitLocation(event)}>
          <input
            className="search-form__input"
            placeholder="Search by City and State or Zip"
            onChange={event => this.updateInput(event)}
            name="location"
            type="text"
            list='cities'
            value={location} />
          <datalist id="cities">
            {
              cities.map((city, index) =>
                <option key={index} value={city} />
              )
            }
          </datalist>
          <button
            className="search-form__button"
            type="submit"
            disabled={!location.trim(' ')}>
            Search
          </button>
        </form>
      </header>
    );
  }
}

Search.propTypes = {
  updateLocation: PropTypes.func,
  location: PropTypes.string,
  autoComplete: PropTypes.func
};

export default Search;