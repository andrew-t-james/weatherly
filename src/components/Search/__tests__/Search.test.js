import React from 'react';
import { shallow } from 'enzyme';
import Search from '../Search.js';

describe('Search Unit Test', () => {
  const mockFn = jest.fn();
  let search;

  beforeEach(() => search =  shallow(<Search updateLocation={mockFn}/>));

  test('it should have a form, input, and button', () => {
    const expectedForm = 1;
    const actualForm = search.find('form').length;

    const expectedButton = 'Search';
    const actualButton = search.find('button').text();

    const disabledButton = true;
    const actualDisabledButton = search.find('button').is('[disabled]');

    const expectedInput = 1;
    const actualInput = search.find('input').length;

    expect(actualForm).toBe(expectedForm);
    expect(actualButton).toBe(expectedButton);
    expect(actualDisabledButton).toBe(disabledButton);
    expect(actualInput).toBe(expectedInput);
  });

  test('Form should have have default state value title is an empty string', () => {
    const expected = '';
    const actual = search.state('location');

    expect(actual).toBe(expected);
  });

  test('it should update state onChange', () => {
    const mockEvent = {
      target: {
        value: 'Louisville',
        name: 'location'
      }
    };

    const expected = 'Louisville';

    search.instance().updateInput(mockEvent);

    const actual = search.state('location');

    expect(actual).toEqual(expected);
  });

  test('it should enable button when State is updated', () => {
    const expected = false;
    const mockEvent = {
      target: {
        value: 'Louisville',
        name: 'location'
      }
    };

    search.instance().updateInput(mockEvent);

    search.setState({
      location: 'Louisville',
    });

    const actual = search.find('button').html().includes('disabled');

    expect(actual).toEqual(expected);
  });

  test('it should enable button when State is updated with valid input', () => {
    const expected = true;
    const mockEvent = {
      target: {
        value: '   ',
        name: 'location'
      }
    };

    search.instance().updateInput(mockEvent);

    search.setState({
      location: '   ',
    });

    const actual = search.find('button').html().includes('disabled');

    expect(actual).toEqual(expected);
  });

  test('it should clear state on submit', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: {
        reset: jest.fn()
      }
    };

    const expected = '';

    search.instance().submitLocation(mockEvent);

    const actual = search.state('location');

    expect(actual).toEqual(expected);
  });

  test('When form submitLocation method is invoked with the correct arguments', () => {
    const expected = 'Louisville';

    const mockEvent = {
      preventDefault: jest.fn(),
      target: {
        reset: jest.fn()
      }
    };


    search.setState({
      location: 'Louisville'
    });

    search.instance().submitLocation(mockEvent);
    expect(mockFn).toHaveBeenCalledWith(expected);
  });
});