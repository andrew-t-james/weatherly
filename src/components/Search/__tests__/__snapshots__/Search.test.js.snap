// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Unit Test it should match snapshot 1`] = `
<header
  className="header"
>
  <h1
    className="header__title"
  >
    Weathrly
  </h1>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <input
      className="search-form__input"
      name="location"
      onChange={[Function]}
      placeholder="Search by City and State or Zip"
      type="text"
      value=""
    />
    <button
      className="search-form__button"
      disabled={true}
      type="submit"
    >
      Search
    </button>
  </form>
</header>
`;
